internal/sidekick/internal/rust/annotate.go

```diff
@@ -991,12 +996,22 @@ func (c *codec) annotateOneOf(oneof *api.OneOf, message *api.Message, model *api
 	qualifiedName := fmt.Sprintf("%s::%s", scope, enumName)
 	relativeEnumName := strings.TrimPrefix(qualifiedName, c.modulePath+"::")
 	structQualifiedName := fullyQualifiedMessageName(message, c.modulePath, model.PackageName, c.packageMapping)
+	scopeInExamples := scope
+	if strings.HasPrefix(scope, c.modulePath+"::") {
+		scopeInExamples = strings.Replace(scope, c.modulePath, fmt.Sprintf("%s::model", c.packageNamespace(model)), 1)
+	}
 
 	bestField := slices.MaxFunc(oneof.Fields, func(f1 *api.Field, f2 *api.Field) int {
 		if f1.Deprecated == f2.Deprecated {
 			if f1.Map == f2.Map {
 				if f1.Repeated == f2.Repeated {
-					return 0
+					if f1.MessageType != nil && f2.MessageType == nil {
```

Comment:
For a separate PR: this function is getting large enough that it needs unit tests.

Reply:
annotate_test.TestOneOfExampleFieldSelection is meant to test just this, if that's not enough I'm happy to add more cases there and/or extract this function to test it directly.